Should g++ and clang++ be interchangeable for a C++ program using std C++ headers?

<!--
Thanks for reporting your issue.
Please fill out the sections below.
-->
Issue summary:
**What is the correct way to setup conda environment for using clang++ as a C++ compiler?**

Consider a conda environment created with the following dependencies:
```
compilers
clangxx
```
and a simple C++ test program that includes some std C++ headers:
```
#include <iostream>
#include <limits>

int main() {
  std::cout << std::boolalpha;
  std::cout << "Minimum value for int: " << std::numeric_limits<int>::min() << '\n';
  return 0;
}
```

When using `g++`, the program compiles fine:
```
g++ numeric_limits.cpp
```
Now switching to `clang++`, the program compilation fails:
```
$ clang++ numeric_limits.cpp
numeric_limits.cpp:1:10: fatal error: 'iostream' file not found
#include <iostream>
         ^~~~~~~~~~
1 error generated.
```
While there are numerous ways to ensure that `clang++` will find the required headers, but **shouldn't `clang++` be able to compile this program as show above?**

<br/>
Some diagnostics:

Clearly, the problem is that `clang++` is not able to find the std C++ headers from the conda environment. In fact,
```
clang++ -E -x c++ - -v < /dev/null
```
<details>

```
clang version 9.0.0 (https://github.com/conda-forge/clangdev-feedstock 007a1ad948eb017506985da76603989f7e02786a)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/pearu/miniconda3/envs/compilers-test/bin
Found candidate GCC installation: /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.4.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.4.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
Found CUDA installation: /usr/local/cuda, version 10.1
 "/home/pearu/miniconda3/envs/compilers-test/bin/clang-9" -cc1 -triple x86_64-unknown-linux-gnu -E -disable-free -disable-llvm-verifier -discard-value-names -main-file-name - -mrelocation-model static -mthread-model posix -mdisable-fp-elim -fmath-errno -masm-verbose -mconstructor-aliases -munwind-tables -fuse-init-array -target-cpu x86-64 -dwarf-column-info -debugger-tuning=gdb -v -resource-dir /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0 -internal-isystem /usr/local/include -internal-isystem /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -fdeprecated-macro -fdebug-compilation-dir /home/pearu/git/Quansight/pearu-sandbox/compilers-test -ferror-limit 19 -fmessage-length 0 -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -fcolor-diagnostics -faddrsig -o - -x c++ -
clang -cc1 version 9.0.0 based upon LLVM 9.0.0 default target x86_64-unknown-linux-gnu
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
...
```
</details>
shows that `clang++` picks up a GCC installation from the system, although the conda GCC installation is listed as a candidate.

**Is the conda environment missing some dependency or what is the reason for this choice?**

conda `g++`, however, picks up the include paths from conda GCC installation correctly:
<details>

```
$ g++ -E -x c++ - -v < /dev/null
Reading specs from /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/specs
COLLECT_GCC=g++
Target: x86_64-conda_cos6-linux-gnu
Configured with: /home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/configure --build=x86_64-build_pc-linux-gnu --host=x86_64-build_pc-linux-gnu --target=x86_64-conda_cos6-linux-gnu --prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built --with-sysroot=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-languages=c,c++,fortran,objc,obj-c++ --with-pkgversion='crosstool-NG 1.23.0.452-d158' --enable-__cxa_atexit --disable-libmudflap --enable-libgomp --disable-libssp --enable-libquadmath --enable-libquadmath-support --enable-libsanitizer --enable-libmpx --with-gmp=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpfr=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-mpc=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --with-isl=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/.build/x86_64-conda_cos6-linux-gnu/buildtools --enable-lto --with-host-libstdcxx='-static-libgcc -Wl,-Bstatic,-lstdc++,-Bdynamic -lm' --enable-threads=posix --enable-target-optspace --enable-plugin --enable-gold --disable-nls --disable-multilib --with-local-prefix=/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot --enable-long-long --enable-default-pie
Thread model: posix
gcc version 7.3.0 (crosstool-NG 1.23.0.452-d158) 
COLLECT_GCC_OPTIONS='-E' '-v' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
 /home/pearu/miniconda3/envs/compilers-test/bin/../libexec/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/cc1plus -E -quiet -v -iprefix /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/ -isysroot /home/pearu/miniconda3/envs/compilers-test/bin/../x86_64-conda_cos6-linux-gnu/sysroot -D_GNU_SOURCE - -mtune=generic -march=x86-64
ignoring duplicate directory "/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0"
ignoring duplicate directory "/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu"
ignoring duplicate directory "/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/backward"
ignoring duplicate directory "/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include"
ignoring nonexistent directory "/home/pearu/miniconda3/envs/compilers-test/bin/../x86_64-conda_cos6-linux-gnu/sysroot/home/conda/feedstock_root/build_artifacts/ctng-compilers_1570984655235/work/gcc_built/x86_64-conda_cos6-linux-gnu/sysroot/include"
ignoring duplicate directory "/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include-fixed"
ignoring duplicate directory "/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/../../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/backward
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/include-fixed
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include
 /home/pearu/miniconda3/envs/compilers-test/bin/../x86_64-conda_cos6-linux-gnu/sysroot/usr/include
End of search list.
# 1 "<stdin>"
# 1 "<built-in>"
# 1 "<command-line>"
# 1 "<stdin>"
COMPILER_PATH=/home/pearu/miniconda3/envs/compilers-test/bin/../libexec/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/:/home/pearu/miniconda3/envs/compilers-test/bin/../libexec/gcc/:/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/
LIBRARY_PATH=/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/:/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/:/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/lib/../lib/:/home/pearu/miniconda3/envs/compilers-test/bin/../x86_64-conda_cos6-linux-gnu/sysroot/lib/../lib/:/home/pearu/miniconda3/envs/compilers-test/bin/../x86_64-conda_cos6-linux-gnu/sysroot/usr/lib/../lib/:/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/lib/:/home/pearu/miniconda3/envs/compilers-test/bin/../x86_64-conda_cos6-linux-gnu/sysroot/lib/:/home/pearu/miniconda3/envs/compilers-test/bin/../x86_64-conda_cos6-linux-gnu/sysroot/usr/lib/
COLLECT_GCC_OPTIONS='-E' '-v' '-shared-libgcc' '-mtune=generic' '-march=x86-64'
(compilers-test) :(compilers-test)pearu@ex:~/git/Quansight/pearu-sandbox/compilers-test$ 
```
</details>

One can force the conda GCC installation using `--gcc-toolchain` option:
<details>

```
$ clang++ --gcc-toolchain=$CONDA_PREFIX -E -x c++ - -v < /dev/null
clang version 9.0.0 (https://github.com/conda-forge/clangdev-feedstock 007a1ad948eb017506985da76603989f7e02786a)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/pearu/miniconda3/envs/compilers-test/bin
Found candidate GCC installation: /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0
Selected GCC installation: /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
Found CUDA installation: /usr/local/cuda, version 10.1
 "/home/pearu/miniconda3/envs/compilers-test/bin/clang-9" -cc1 -triple x86_64-unknown-linux-gnu -E -disable-free -disable-llvm-verifier -discard-value-names -main-file-name - -mrelocation-model static -mthread-model posix -mdisable-fp-elim -fmath-errno -masm-verbose -mconstructor-aliases -munwind-tables -fuse-init-array -target-cpu x86-64 -dwarf-column-info -debugger-tuning=gdb -v -resource-dir /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0 -internal-isystem /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0 -internal-isystem /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu -internal-isystem /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/backward -internal-isystem /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/sysroot/usr/local/include -internal-isystem /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0/include -internal-externc-isystem /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/sysroot/include -internal-externc-isystem /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/sysroot/usr/include -fdeprecated-macro -fdebug-compilation-dir /home/pearu/git/Quansight/pearu-sandbox/compilers-test -ferror-limit 19 -fmessage-length 0 -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -fcolor-diagnostics -faddrsig -o - -x c++ -
clang -cc1 version 9.0.0 based upon LLVM 9.0.0 default target x86_64-unknown-linux-gnu
ignoring nonexistent directory "/home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/sysroot/usr/local/include"
ignoring nonexistent directory "/home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/sysroot/include"
#include "..." search starts here:
#include <...> search starts here:
 /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0
 /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu
 /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/backward
 /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0/include
 /home/pearu/miniconda3/envs/compilers-test/lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/sysroot/usr/include
End of search list.
```
</details>
and then the test program compiles fine.

**Why `--gcc-toolchain=<conda prefix>` would not be a default for `clang++`?**

<br/>
<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /home/pearu/miniconda3/envs/compilers-test:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
binutils-meta             1.0.4                         0    conda-forge
binutils_impl_linux-64    2.31.1               h7fc9f1b_2    conda-forge
binutils_linux-64         2.31.1              h6176602_12    conda-forge
c-compiler                1.0.4                h516909a_0    conda-forge
ca-certificates           2019.9.11            hecc5488_0    conda-forge
certifi                   2019.9.11                py38_0    conda-forge
clang                     9.0.0           default_hde54327_3    conda-forge
clang-tools               9.0.0           default_hde54327_3    conda-forge
clangdev                  9.0.0           default_hde54327_3    conda-forge
clangxx                   9.0.0           default_h6bfbf51_3    conda-forge
compilers                 1.0.4                         0    conda-forge
cxx-compiler              1.0.4                hc9558a2_0    conda-forge
fortran-compiler          1.0.4                he991be0_0    conda-forge
gcc_impl_linux-64         7.3.0                habb00fd_2    conda-forge
gcc_linux-64              7.3.0               h553295d_12    conda-forge
gfortran_impl_linux-64    7.3.0                hdf63c60_2    conda-forge
gfortran_linux-64         7.3.0               h553295d_12    conda-forge
gxx_impl_linux-64         7.3.0                hdf63c60_2    conda-forge
gxx_linux-64              7.3.0               h553295d_12    conda-forge
libclang                  9.0.0           default_hde54327_3    conda-forge
libffi                    3.2.1             he1b5a44_1006    conda-forge
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_2    conda-forge
libllvm9                  9.0.0                hc9558a2_3    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
llvm-tools                9.0.0                hc9558a2_3    conda-forge
llvmdev                   9.0.0                hc9558a2_3    conda-forge
ncurses                   6.1               hf484d3e_1002    conda-forge
openssl                   1.1.1d               h516909a_0    conda-forge
pip                       19.3.1                   py38_0    conda-forge
python                    3.8.0                h357f687_3    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
setuptools                41.6.0                   py38_1    conda-forge
sqlite                    3.30.1               hcee41ef_0    conda-forge
tk                        8.6.9             hed695b0_1003    conda-forge
wheel                     0.33.6                   py38_0    conda-forge
xz                        5.2.4             h14c3975_1001    conda-forge
zlib                      1.2.11            h516909a_1006    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
     active environment : compilers-test
    active env location : /home/pearu/miniconda3/envs/compilers-test
            shell level : 2
       user config file : /home/pearu/.condarc
 populated config files : 
          conda version : 4.7.12
    conda-build version : not installed
         python version : 3.7.4.final.0
       virtual packages : __cuda=10.1
       base environment : /home/pearu/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/pearu/miniconda3/pkgs
                          /home/pearu/.conda/pkgs
       envs directories : /home/pearu/miniconda3/envs
                          /home/pearu/.conda/envs
               platform : linux-64
             user-agent : conda/4.7.12 requests/2.22.0 CPython/3.7.4 Linux/4.15.0-66-generic ubuntu/18.04.3 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

```
</details>

Thanks, @isuruf ! That helped but:
```
$ clang++ numeric_limits.cpp
In file included from numeric_limits.cpp:1:
In file included from /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/iostream:38:
In file included from /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu/bits/c++config.h:533:
/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu/bits/os_defines.h:39:10: fatal error: 'features.h' file not found
#include <features.h>
         ^~~~~~~~~~~~
1 error generated.
```

It also appears that the issue may depend on the system. I can reproduce the issue on Ubuntu 18.04.3 LTS but not on Ubuntu 16.04.5 LTS.

Here:
```
$ clang++ numeric_limits.cpp -v
clang version 9.0.0 (https://github.com/conda-forge/clangdev-feedstock 007a1ad948eb017506985da76603989f7e02786a)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/pearu/miniconda3/envs/compilers-test/bin
Found candidate GCC installation: /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0
Selected GCC installation: /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
 "/home/pearu/miniconda3/envs/compilers-test/bin/clang-9" -cc1 -triple x86_64-unknown-linux-gnu -emit-obj -mrelax-all -disable-free -disable-llvm-verifier -discard-value-names -main-file-name numeric_limits.cpp -mrelocation-model static -mthread-model posix -mdisable-fp-elim -fmath-errno -masm-verbose -mconstructor-aliases -munwind-tables -fuse-init-array -target-cpu x86-64 -dwarf-column-info -debugger-tuning=gdb -v -resource-dir /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0 -isysroot /x86_64-conda_cos6-linux-gnu/sysroot -internal-isystem /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0 -internal-isystem /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu -internal-isystem /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/backward -internal-isystem /x86_64-conda_cos6-linux-gnu/sysroot/usr/local/include -internal-isystem /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0/include -internal-externc-isystem /x86_64-conda_cos6-linux-gnu/sysroot/include -internal-externc-isystem /x86_64-conda_cos6-linux-gnu/sysroot/usr/include -fdeprecated-macro -fdebug-compilation-dir /home/pearu/git/Quansight/pearu-sandbox/compilers-test -ferror-limit 19 -fmessage-length 0 -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -fcolor-diagnostics -faddrsig -o /tmp/numeric_limits-59bade.o -x c++ numeric_limits.cpp
clang -cc1 version 9.0.0 based upon LLVM 9.0.0 default target x86_64-unknown-linux-gnu
ignoring nonexistent directory "/x86_64-conda_cos6-linux-gnu/sysroot/usr/local/include"
ignoring nonexistent directory "/x86_64-conda_cos6-linux-gnu/sysroot/include"
ignoring nonexistent directory "/x86_64-conda_cos6-linux-gnu/sysroot/usr/include"
#include "..." search starts here:
#include <...> search starts here:
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu
 /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/backward
 /home/pearu/miniconda3/envs/compilers-test/lib/clang/9.0.0/include
End of search list.
In file included from numeric_limits.cpp:1:
In file included from /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/iostream:38:
In file included from /home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu/bits/c++config.h:533:
/home/pearu/miniconda3/envs/compilers-test/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/include/c++/7.3.0/x86_64-conda_cos6-linux-gnu/bits/os_defines.h:39:10: fatal error: 'features.h' file not found
#include <features.h>
         ^~~~~~~~~~~~
1 error generated.

```

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,7 +1,7 @@
 {% set version = "9.0.0" %}
 {% set major_version = version.split(".")[0] %}
 {% set sha256 = "7ba81eef7c22ca5da688fdf9d88c20934d2d6b40bfe150ffd338900890aa4610" %}
-{% set build_number = 5 %}
+{% set build_number = 4 %}
 
 package:
   name: clang_packages

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,7 +1,7 @@
 {% set version = "9.0.0" %}
 {% set major_version = version.split(".")[0] %}
 {% set sha256 = "7ba81eef7c22ca5da688fdf9d88c20934d2d6b40bfe150ffd338900890aa4610" %}
-{% set build_number = 4 %}
+{% set build_number = 5 %}
 
 package:
   name: clang_packages
@@ -173,8 +173,11 @@ outputs:
       run:
         - {{ pin_subpackage("clang", exact=True) }}
     test:
+      files:
+        - mytest.cxx
       commands:
         - clang++ --version
+        - clang++ -v -c mytest.cxx
 
   - name: clang-tools
     script: install_clang_tools.sh    # [unix]

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -173,6 +173,8 @@ outputs:
       run:
         - {{ pin_subpackage("clang", exact=True) }}
     test:
+      requires:
+        - {{ compiler("cxx") }}
       files:
         - mytest.cxx
       commands:

--- a/recipe/mytest.cxx
+++ b/recipe/mytest.cxx
@@ -1,0 +1,9 @@
+#include <iostream>
+int main()
+{
+  int result = 0;
+  for (int i = 0; i < 20; ++i)
+    result += i;
+  std::cout << "result=" << result << std::endl;
+  return 0;
+}

